packages/@sanity/types/src/schema/definition/type/array.ts

```diff
@@ -10,6 +10,8 @@ import {
 } from '../schemaDefinition'
 import {type BaseSchemaDefinition, type SearchConfiguration, type TitledListValue} from './common'
 
+export type {InsertMenuOptions}
```

Comment:
By re-exporting this type it ensures that in projects that only have a dependency on `@sanity/types`, and otherwise don't have `@sanity/insert-menu`, get proper typings on `ArrayOptions['insertMenu']`: https://github.com/sanity-io/sanity/blob/163966da487072a44d0707fe055c59901b084a18/packages/%40sanity/types/src/schema/definition/type/array.ts#L26

The `@sanity/types` package currently has minimal deps, which is awesome: https://www.npmjs.com/package/@sanity/types?activeTab=dependencies

The `@sanity/insert-menu` is of course intended to be used for far more than typing, it's self contained and among other things depend on `@sanity/ui`: https://www.npmjs.com/package/@sanity/insert-menu?activeTab=dependencies

So by inlining its typings we get the best of both, self contained typings, fully compatible with the types used by `sanity`, without adding to the dependency burden of projects who only wishes to use `@sanity/types`:

```diff
diff --git a/lib/index.d.ts b/lib/index.d.ts
index v3.52.3-update-pkg-utils.7..v3.52.3-update-pkg-utils.17 100644
--- a/lib/index.d.ts
+++ b/lib/index.d.ts
@@ -1,6 +1,7 @@
+/// <reference types="react" />
+
 import {ComponentType} from 'react'
 import {ElementType} from 'react'
-import {InsertMenuOptions} from '@sanity/insert-menu'
 import {ReactElement} from 'react'
 import {ReactNode} from 'react'
 import {SanityClient} from '@sanity/client'
@@ -1548,6 +1549,33 @@
   }
 }
 
+/** @alpha This API may change */
+export declare interface InsertMenuOptions {
+  /**
+   * @defaultValue `'auto'`
+   * `filter: 'auto'` automatically turns on filtering if there are more than 5
+   * schema types added to the menu.
+   */
+  filter?: 'auto' | boolean
+  groups?: Array<{
+    name: string
+    title?: string
+    of?: Array<string>
+  }>
+  /** defaultValue `true` */
+  showIcons?: boolean
+  /** @defaultValue `[{name: 'list'}]` */
+  views?: Array<
+    | {
+        name: 'list'
+      }
+    | {
+        name: 'grid'
+        previewImageUrl?: (schemaTypeName: string) => string | undefined
+      }
+  >
+}
+
 /** @internal */
 export declare type InsertPatch =
   | {

```